go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -446,21 +431,29 @@ func (bsnc *BaseSecondaryNetworkController) removePodForSecondaryNetwork(pod *ka
 		portInfoMap = map[string]*lpInfo{}
 	}
 
-	activeNetwork, err := bsnc.getActiveNetworkForNamespace(pod.Namespace)
-	if err != nil {
-		return fmt.Errorf("failed looking for the active network at namespace '%s': %w", pod.Namespace, err)
-	}
+	var alreadyProcessed bool
 	for nadName := range podNetworks {
 		if !bsnc.HasNAD(nadName) {
```

Comment:
self-note this is what helps here, if this pod is not part of this network then we don't process it, so that's good
@jcaamano or @maiqueb do we know why we added the whole "Active network"/"GetPodNADToNetworkMappingWithActiveNetwork" logic here without resusing the logic we had already for secondary networks to determing if this pod is part of this network controller or not..?